engine/access/ingestion/collections/syncer.go

```diff
@@ -100,53 +94,61 @@ func NewSyncer(
 		indexer:             collectionIndexer,
 		execDataSyncer:      execDataSyncer,
 
-		collectionCatchupTimeout:        DefaultCollectionCatchupTimeout,
-		collectionCatchupDBPollInterval: DefaultCollectionCatchupDBPollInterval,
-		missingCollsForBlockThreshold:   DefaultMissingCollsForBlockThreshold,
-		missingCollsForAgeThreshold:     DefaultMissingCollsForAgeThreshold,
-		missingCollsRequestInterval:     DefaultMissingCollsRequestInterval,
+		collectionCatchupTimeout:          DefaultCollectionCatchupTimeout,
+		collectionCatchupDBPollInterval:   DefaultCollectionCatchupDBPollInterval,
+		missingCollectionRequestThreshold: DefaultMissingCollectionRequestThreshold,
+		missingCollectionRequestInterval:  DefaultMissingCollectionRequestInterval,
 	}
 }
 
 // WorkerLoop is a [component.ComponentWorker] that continuously monitors for missing collections, and
 // requests them from the network if needed. It also performs an initial collection catchup on startup.
 func (s *Syncer) WorkerLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
-	requestCtx, cancel := context.WithTimeout(ctx, s.collectionCatchupTimeout)
-	defer cancel()
-
-	// attempt to download all known missing collections on start-up.
-	err := s.requestMissingCollectionsBlocking(requestCtx)
-	if err != nil {
-		if ctx.Err() != nil {
-			s.log.Error().Err(err).Msg("engine shutdown while downloading missing collections")
+	// Block marking the component ready until either the first run of the missing collections catchup
+	// completes, or `collectionCatchupTimeout` expires. This improves the user experience by preventing
+	// the Access API from starting while the initial catchup completes. The intention is to avoid
+	// returning NotFound errors immediately after startup for data that should be available, but has
+	// not yet been indexed.
+	// TODO (peter): consider removing this. I'm not convinced that this provides much value since in
+	// the common case (node restart), the block finalization would also pause so the node should not
+	// become farther behind in terms of collections.
+	readyChan := make(chan struct{})
+	go func() {
+		select {
+		case <-ctx.Done():
 			return
+		case <-readyChan:
+		case <-time.After(s.collectionCatchupTimeout):
 		}
+		ready()
+	}()
 
-		if !errors.Is(err, context.DeadlineExceeded) {
-			ctx.Throw(fmt.Errorf("error downloading missing collections: %w", err))
+	requestCollectionsTicker := time.NewTicker(s.missingCollectionRequestInterval)
+	defer requestCollectionsTicker.Stop()
+
+	initialCatchupComplete := false
+	for {
+		err := s.requestMissingCollections(ctx, !initialCatchupComplete)
```

Comment:
Changes looks good. 

I have one concern for syncer's requestMissingCollections :

The syncing asks the requester to request missing collections, after a few seconds, the execution data indexer indexed the execution data, then the syncer is able to index the collections, and the indexer will move forward the last full block height. However, the requester didn't know about the fact that the collections have been received, so I think over time, it will build up a long list of zombie collections in the fetch list, we need to check how the requester deals with it, I'm afraid those zombie collections will affect the actual missing collection requests from being sent.

Reply:
that's a good point. the requester should eventually download the collections, but it is wasted effort. it may be worth adding a timeout for AN collection syncing